### PR TITLE
Upgrade flit

### PR DESCRIPTION
**Pull request by JPeterMugaas**
*) Use pyproject2setppy to compile flit
*) include documentation in dirhtml form and in man form
*) List all packages that were listed when I ran a bootstrap stcript
*) Upgrade python-py
*) Add new packages,  mingw-w64-python-pygments-github-lexers and mingw-w64-python-sphinxcontrib-github-alt to for syntax-highlighting in documentation

### CODE REVIEW

**Comment by Biswa96** [comment]
Separate commits for each packages would be easier to review.

File: mingw-w64-python-flit/PKGBUILD [hunk 1]
```diff
@@ -1,89 +1,158 @@
...
-_pyname=flit
 _realname=flit
 pkgbase=mingw-w64-python-${_realname}
-pkgname=(
-    "${MINGW_PACKAGE_PREFIX}-python-${_realname}"
-    "${MINGW_PACKAGE_PREFIX}-python-${_realname}-core"
-)
-pkgver=3.2.0
-pkgrel=2
+pkgbase=${MINGW_PACKAGE_PREFIX}-python-$_realname
```
Comment (jeremyd2019) [inline mingw-w64-python-flit/PKGBUILD:5]:
```suggestion
```

File: mingw-w64-python-flit/PKGBUILD [hunk 2]
```diff
@@ -1,89 +1,158 @@
...
+pkgname=(${MINGW_PACKAGE_PREFIX}-python-$_realname{,-core})
+pkgver=3.5.1
+pkgrel=1
 pkgdesc='Simplified packaging of Python modules (mingw-w64)'
 arch=('any')
-mingw_arch=('mingw32' 'mingw64' 'ucrt64' 'clang64' 'clang32' 'clangarm64')
```
Comment (jeremyd2019) [inline mingw-w64-python-flit/PKGBUILD:14]: mingw_arch is necessary, please don't remove

File: mingw-w64-python-flit/PKGBUILD [hunk 3]
```diff
@@ -1,89 +1,158 @@
...
-  install -Dm644 LICENSE "${pkgdir}${MINGW_PREFIX}/share/licenses/python-${_realname}-core/LICENSE"
+  # installing tests to a subdir of your project might not conflict the toplevel, but
+  # don't do it anyway...
+  rm -rf "$pkgdir"${MINGW_PREFIX}/lib/python*/site-packages/flit_core/tests/
+  
+  find "$pkgdir" -name "*.json" -exec \
+    sed -e "s|$srcdir||g" -i {} \;
 }
 
-# template start; name=mingw-w64-splitpkg-wrappers; version=1.0;
```
Comment (jeremyd2019) [inline mingw-w64-python-flit/PKGBUILD:80]: The template comments need to stay too

File: mingw-w64-python-pygments-github-lexers/PKGBUILD [hunk 4]
```diff
@@ -0,0 +1,44 @@
+# Maintainer: J. Peter Mugaas <jpmugaas@suddenlink.net>
+
+_realname=pygments-github-lexers
+pkgbase=mingw-w64-python-${_realname}
+pkgname=("${MINGW_PACKAGE_PREFIX}-python-${_realname}")
+pkgver=0.0.5
+pkgrel=1
+pkgdesc="Pygments Github custom lexers. (mingw-w64)"
+arch=('any')
```
Comment (jeremyd2019) [inline mingw-w64-python-pygments-github-lexers/PKGBUILD:9]: add mingw_arch

File: mingw-w64-python-sphinxcontrib-github-alt/PKGBUILD [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+#
+# https://github.com/jupyter/sphinxcontrib_github_alt/archive/refs/tags/1.2.tar.gz
+#
+_realname=sphinxcontrib-github-alt
+pkgbase=mingw-w64-python-${_realname}
+pkgname=("${MINGW_PACKAGE_PREFIX}-python-${_realname}")
+pkgver=1.2
+pkgrel=1
+pkgdesc="Link to GitHub issues, pull requests, commits and users from Sphinx docs. (mingw-w64)"
+arch=('any')
```
Comment (jeremyd2019) [inline mingw-w64-python-sphinxcontrib-github-alt/PKGBUILD:11]: add mingw_arch

File: mingw-w64-python-flit/PKGBUILD [hunk 6]
```diff
@@ -1,89 +1,159 @@
...
-# vim: set ft=bash :
-
-# generate wrappers
 for _name in "${pkgname[@]}"; do
   _short="package_${_name#${MINGW_PACKAGE_PREFIX}-}"
   _func="$(declare -f "${_short}")"
   eval "${_func/#${_short}/package_${_name}}"
 done
-# template end;
+
```
Comment (jeremyd2019) [inline mingw-w64-python-flit/PKGBUILD:159]:
```suggestion
# template end;
```

**Comment by jeremyd2019** [comment]
There are a lot of makedepends on flit now, and they seem to overlap with checkdepends.  Are you sure these are all necessary in makedepends (if check is enabled, makepkg will install both the makedepends and checkdepends, otherwise just the makedepends).

**Pull request merged**